### ISSUE TITLE
Please remove dependency on os-homedir

### ISSUE THREAD

**Issue by pravi** [npm/osenv#11]
Its has serious unfixed issues https://github.com/sindresorhus/os-homedir/issues/4 expand-tilde already removed this dependency.

https://github.com/jonschlinkert/expand-tilde/commit/e13f94abd49f766688dc6a93fc483384a5f15868

**Comment by iarna** [comment]
As of this writing `osenv` is a dependency of `npm` and `npm` supports node 0.10. Node 0.10 does not support `os.homedir` as such, _some_ shim is required and without that we can't just ditch `os-homedir`. If you _have_ a better shim for it then I suspect you won't have trouble getting a PR for `os-homedir` accepted to use it.

**Comment by pravi** [comment]
@iarna, fair enough you want to support older nodejs versions, but are you saying its the responaibility of random bug reporters to fix critical bugs in your own code? Or are you saying the bug in os-homedir somehow does not affect your code?

**Comment by pravi** [comment]
As for my usecase, debian has on older osenv without os-homedir dependency and since we already have a newer nodejs, we don't need os-homedir. So I don't need to fix os-homedir. It is upto you, how you want to keep your code in good shape.

**Comment by welladamm** [comment]
IMHO, this case should be taken of specially. We can't rely on an underlying implementation known to be defected. Can we implement our own polyfill on the npm side?

**Comment by iarna** [comment]
I don't see this as critical as you seem to (obviously seeing as I'm just getting back to this now). We have a very, very small team and so our work is all about triage. I wasn't insisting that you do any work, just telling you what we could and could not accept.

@welladamm We could patch `os-homedir` itself if anyone could come up with better behavior. Thus far I've seen no proposals. (The suggestions in the linked issue are all about C-system calls that are unavailable in Node. Any solution must be pure-JS.)
